tests/unittest.py

```diff
@@ -307,6 +408,20 @@ def assertTransactionConsensusValid(self, tx):
                 self.assertTrue(meta.voided_by)
                 self.assertTrue(parent_meta.voided_by.issubset(meta.voided_by))
 
+    def assertSynced(self, node_sync):
+        """Check "synced" status of p2p-manager, uses self._enable_sync_vX to choose which check to run."""
+        enable_sync_v1, enable_sync_v2 = self._syncVersionFlags()
```

Comment:
Should we add a method `isSynced()` to the node_sync's class?

Reply:
The sync algorithm abstract class already has a `is_synced` method, but it has a different meaning. Since this check currently only has a use for tests, I'd rather keep it in tests only for now.

Reply:
Renamed to `assertSyncedProgress`. All use cases also do a `assertTipsEqual` which will cover the mempools are equals.